### PR TITLE
Introduce jms connectors

### PR DESCRIPTION
**Pull request by gurbuzali**

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/StreamJmsP.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        session = sessionFn.apply(connection);
+        consumer = consumerFn.apply(session);
+    }
+
+    @Override
+    public boolean complete() {
+        if (!emitPendingItem()) {
+            return false;
+        }
+        Message message = uncheckCall(() -> consumer.receiveNoWait());
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/StreamJmsP.java]: wouldn't it make sense to consume more than one msg inside one `complete` call?

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJmsP.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import static com.hazelcast.jet.impl.util.Util.uncheckCall;
+import static com.hazelcast.jet.impl.util.Util.uncheckRun;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Private API. Access via {@link
+ * com.hazelcast.jet.core.processor.SinkProcessors#writeJmsQueueP} or
+ * {@link com.hazelcast.jet.core.processor.SinkProcessors#writeJmsTopicP}
+ */
+public final class WriteJmsP {
```
[the remark this answers is not in the document]
Comment (gurbuzali) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJmsP.java]: it is using `WriteBufferedP` which is not cooperative

[the remark this answers is not in the document]
Comment (viliam-durina) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJmsP.java]: ok, i retract

**Comment by gurbuzali** [comment]
verify

**Comment by gurbuzali** [comment]
LoggerSinkTest is failing repeatedly, created an issue https://github.com/hazelcast/hazelcast-jet/issues/876

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJmsP.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        @Override
+        public Collection<? extends Processor> get(int count) {
+            DistributedFunction<Processor.Context, JmsContext> createFn = jet -> {
+                Session session = sessionF.apply(connection);
+                Destination destination =
+                        uncheckCall(() -> isTopic ? session.createTopic(name) : session.createQueue(name));
+                MessageProducer producer = uncheckCall(() -> session.createProducer(destination));
+                return new JmsContext(session, producer);
+            };
+            DistributedBiConsumer<JmsContext, T> onReceiveFn = (buffer, item) -> {
```
Comment (lukasherman) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJmsP.java]: `buffer` variable name is misleading here (also occurs below), should be `jmsContext`.

Comment (viliam-durina) [reply hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJmsP.java]: You're right, but this PR is merged. Feel free to submit a PR :)

**Pull request merged**